roles/openshift_metering/tasks/install.yaml

```diff
@@ -0,0 +1,94 @@
+---
+- name: Create temp directory for doing work in on target
+  command: mktemp -td openshift-metering-ansible-XXXXXX
+  register: mktemp
+  changed_when: False
+
+- name: Ensures {{ mktemp.stdout }}/metering-manifests directories exists
+  file:
+    path: "{{ item }}"
+    state: directory
+  with_items:
+  - "{{ mktemp.stdout }}/metering-manifests/operator"
+
+- name: Copy operator files to temp directory
+  copy:
+    src: "{{ item }}"
+    dest: "{{ mktemp.stdout }}/metering-manifests/operator/{{ item | basename }}"
+  with_fileglob:
+  - "files/operator/*.yaml"
+
+- name: Update Metering Helm Operator deployment image
+  yedit:
+    src: "{{ mktemp.stdout }}/metering-manifests/operator/metering-helm-operator-deployment.yaml"
+    edits:
+    - key: spec.template.spec.containers[0].image
+      value: "{{ openshift_metering_operator_image }}"
+    - key: spec.template.spec.containers[1].image
+      value: "{{ openshift_metering_operator_image }}"
+  when: openshift_metering_operator_image != ""
+
+- name: Update Metering config
+  yedit:
+    src: "{{ mktemp.stdout }}/metering-manifests/operator/metering.yaml"
+    edits:
+    - key: spec
+      value: "{{ openshift_metering_config }}"
+  when: openshift_metering_config != None
+
+- name: "Create {{ __openshift_metering_namespace }} project"
+  oc_project:
+    state: present
+    name: "{{ __openshift_metering_namespace }}"
+    description: Openshift Metering
+    node_selector: ""
+
+- name: Create metering namespace viewer cluster role
+  oc_clusterrole:
+    state: present
+    name: "openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+    rules:
+    - apiGroups: [""]
+      resources: ["namespaces"]
+      verbs: ["get"]
+
+- name: Create namespace viewer clusterrolebinding
+  oc_obj:
+    state: present
+    kind: ClusterRoleBinding
+    name: "openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+    content:
+      path: "{{ mktemp.stdout }}/openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+      data:
+        apiVersion: v1
+        kind: ClusterRoleBinding
+        metadata:
+          name: "openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+        roleRef:
+          apiGroup: rbac.authorization.k8s.io
+          kind: ClusterRole
+          name: "openshift-metering-namespace-viewer-{{ __openshift_metering_namespace }}"
+        subjects:
+        - kind: ServiceAccount
+          name: metering
+          namespace: "{{ __openshift_metering_namespace }}"
+
+- name: Install Metering manifests
+  oc_obj:
+    state: present
+    kind: CustomResourceDefinition
+    name: "{{ obj.metadata.name }}"
+    namespace: "{{ __openshift_metering_namespace }}"
+    files:
```

Comment:
`oc_obj` can't read straight from file, instead
```
    content:
      path: /tmp/{{ obj.metadata.name }}
      data: '{{ obj }}'
```
should be used

Reply:
I tried that and it made no difference.

Reply:
Ah, seems I was wrong, the message was just standard SFTP probing to see if the file exists. So I tried the above change, and I get the following:

```
TASK [openshift_metering : Install Metering manifests] *************************
task path: /usr/share/ansible/openshift-ansible/roles/openshift_metering/tasks/install.yaml:76
Monday 09 July 2018  18:01:02 +0000 (0:00:02.281)       0:00:48.741 ***********
 [WARNING]: Unable to find '/tmp/openshift-metering-ansible-dtCooq/metering-
manifests/operator' in expected paths (use -vvvvv to see paths)

```

Reply:
So I've made progress I tried using `files` for the operator manifests being put into a tempdir but it seems to be failing because lookup plugins run locally, not on the remote host. I'm reading about this and it seems like I'll have get the contents of each file and store them in a variable so I can avoid using `loopup('file', item)` and so I can avoid using `with_fileglob` since both run locally instead of remotely. I also tried `delegate_to: "{{ groups.oo_first_master.0 }}"` but I read lookup plugins don't obey delegation.

```
TASK [openshift_metering : Install Metering manifests] *************************
Monday 09 July 2018  18:24:27 +0000 (0:00:01.906)       0:00:50.520 ***********
 [WARNING]: Unable to find '/tmp/openshift-metering-ansible-tHiet7/metering-
manifests/operator' in expected paths (use -vvvvv to see paths)
changed: [ci-chancez-4-ig-m-gl55] => (item=/usr/share/ansible/openshift-ansible/roles/openshift_metering/files/crds/report.crd.yaml)
changed: [ci-chancez-4-ig-m-gl55] => (item=/usr/share/ansible/openshift-ansible/roles/openshift_metering/files/crds/storagelocation.crd.yaml)
ok: [ci-chancez-4-ig-m-gl55] => (item=/usr/share/ansible/openshift-ansible/roles/openshift_metering/files/crds/metering.crd.yaml)
changed: [ci-chancez-4-ig-m-gl55] => (item=/usr/share/ansible/openshift-ansible/roles/openshift_metering/files/crds/reportgenerationquery.crd.yaml)
changed: [ci-chancez-4-ig-m-gl55] => (item=/usr/share/ansible/openshift-ansible/roles/openshift_metering/files/crds/scheduledreport.crd.yaml)
changed: [ci-chancez-4-ig-m-gl55] => (item=/usr/share/ansible/openshift-ansible/roles/openshift_metering/files/crds/reportprometheusquery.crd.yaml)
changed: [ci-chancez-4-ig-m-gl55] => (item=/usr/share/ansible/openshift-ansible/roles/openshift_metering/files/crds/prestotable.crd.yaml)
changed: [ci-chancez-4-ig-m-gl55] => (item=/usr/share/ansible/openshift-ansible/roles/openshift_metering/files/crds/reportdatasource.crd.yaml)
```